cortexlab/+git/runGitCmd.m

```diff
@@ -0,0 +1,92 @@
+function [exitCode, cmdOut] = runGitCmd(cmd, varargin)
+%GIT.RUNGITCMD runs git commands in MATLAB.
+%
+% Inputs:
+%   `cmd`: A cellstr array of git command(s) to run.
+%   `dir`: An optional string name-value paired argument which specifies
+%   the directory in which to run the command (default is `pwd`)
+%   `echo`: An optional boolean name-value paired argument which specifies
+%   whether to display command output on the MATLAB command window (default
+%   is `true`).
+%
+% Outputs:
+%   `exitCode`: A flag array indicating whether each command in `cmd` was 
+%   run succesfully (0) or there was an error (1).
+%   `cmdOut`: A cellstr array of the output of `cmd`.
+%
+% Example: Get the status and log for the git repository in the working 
+% folder, display the commands' outputs in MATLAB, and save the command
+% outputs in the `cmdOut` cellstr:
+%   [exitCode, cmdOut] = git.runGitCmd({'status', 'log -n 1'});
+%
+% Example: Stash the working changes in the Rigbox repository without
+% displaying the command's output:
+%   rigboxPath = which('addRigboxPaths');
+%   exitCode = git.runGitCmd({'stash push -m "WIP..."'}, 'dir', rigboxPath,... 
+%                             'echo', false);
+
+%% set-up 
+% Make sure `cmd` is a cellstr
+if ~iscellstr(cmd) %#ok<ISCLSTR>
+  error('runGitCmd:invalidInputArg', ['%s requires the first input arg to '...
+        'be a cellstr']);
+end
+% When this function terminates, return to the working directory.
+origDir = pwd;
+cleanup = onCleanup(@() cd(origDir));
+
+% Define default input args in a struct.
+argS.dir = pwd; 
+argS.echo = true;
+
+% Get input args and reshape into two rows (into name-value pairs):
+% 1st row = arg names, 2nd row = arg values. 
+pairedArgs = reshape(varargin,2,[]);
```

Comment:
there's a burgbox function for Name-Value paired args

Reply:
`namedArg`

Reply:
I actually don't like `namedArg` for this type of purpose: it involves repetitive `if, else, end` blocks for each potential name-value pair (e.g. see [sig.timeplot](https://github.com/cortex-lab/signals/blob/48ac0162ce058e70fb148fea0a11acb04f539e49/%2Bsig/timeplot.m#L44-L70)). I like keeping this struct version of things, and will want to adjust functions that use `namedArg` in the manner that `sig.timeplot` does to this struct way of dealing with name-value pairs